infrastructure/src/main/java/org/corfudb/infrastructure/management/ReconfigurationEventHandler.java

```diff
@@ -56,6 +59,28 @@ public static boolean handleFailure(@Nonnull IReconfigurationHandlerPolicy failu
         }
     }
 
+    /**
+     * Calculate a reasonable workflow timeout
+     *
+     * @param runtime Connected corfu runtime instance
+     * @return A reasonable timeout to complete state transfer and rebuild logging unit.
+     */
+    private static Duration getReasonableWorkflowTimeout(CorfuRuntime runtime) {
+
+        // Try to estimate a reasonable timeout to rebuild the logging unit
+        Token trimMark = runtime.getAddressSpaceView().getTrimMark();
+        long tail = runtime.getAddressSpaceView().getAllTails().getLogTail();
+
+        long rangeToReplicate = tail - trimMark.getSequence();
+        // Since the orchestrator client and the fault detector client use
+        // the same configuration its reasonable to use these arguments.
+        // TODO(Maithem): AddNode should use a similar mechanism to set the timeout
+        long numSections = rangeToReplicate / runtime.getParameters().getBulkReadSize();
+        long rpcTimeout = (long) (runtime.getParameters().getRequestTimeout().toMillis() * 1.5);
+        long timeoutInMs = Math.max(DEFAULT_HEAL_TIMEOUT.toMillis(), numSections * rpcTimeout);
```

Comment:
there should be logging somewhere to tell what is the new workflow timeout.

Reply:
This is logged in the consumers of this method.